pkg/record/record.go

```diff
@@ -6,6 +6,6 @@ package record
 
 type Record interface {
 	Validate() error
-	Parse(line string) error
+	Parse(line string) (int, error)
```

Comment:
Used read size 
this is template code 

```	// RecordCode
	if h.RecordCode, size, err = util.ReadField(util.GetField(line, read)); err != nil {
		return 0, fmt.Errorf(fmt.Sprintf(aiParseErrorFmt, "RecordCode"))
	} else {
		read += size
	}
```

Reply:
Right; it's being used inside the `Parse` methods, and then being returned, but I don't see any callers of `Parse` that do anything with the returned value rather than just throwing it out. Seems like it's not necessary to return in that case.

Reply:
Read variable has changed by adding size variable, will indicate next field

Reply:
Like this 
```
       if h.RecordCode, readSize, err = util.ReadField(util.GetField(line, nextPoint)); err != nil {
		return 0, fmt.Errorf(fmt.Sprintf(aiParseErrorFmt, "RecordCode"))
	} else {
		nextPoint += readSize
	}
```

Reply:
```
func GetField(line string, position int) string {

	if position < len(line) {
		return line[position:]
	}

	return ""
}
```
Above function used instead of line[read:] to prevent panic error

```
	// RecordCode
	if h.RecordCode, size, err = util.ReadField(line[read:]); err != nil {
		return 0, fmt.Errorf(fmt.Sprintf(aiParseErrorFmt, "RecordCode"))
	} else {
		read += size
	}
	
```

Reply:
changed 

```
	// RecordCode
	if h.RecordCode, size, err = util.ReadField(line, read); err != nil {
		return 0, fmt.Errorf(fmt.Sprintf(aiParseErrorFmt, "RecordCode"))
	} else {
		read += size
	}
```

Reply:
That's fine, but all I was saying is I don't think we need to return it from `Parse`, since nothing uses it (aside from tests, but surely we can test the correct behavior without exposing that detail).